### PR TITLE
core: always calculate the next restart interval

### PR DESCRIPTION
**Pull request by YHNdnzj**
Let's always calculate the next restart interval since that's more useful.

For that, we add 1 to s->n_restarts unconditionally, and change RestartUSecCurrent property to RestartUSecNext.

### CODE REVIEW

File: man/org.freedesktop.systemd1.xml [hunk 1]
```diff
@@ -2568,7 +2568,7 @@ node /org/freedesktop/systemd1/unit/avahi_2ddaemon_2eservice {
       @org.freedesktop.DBus.Property.EmitsChangedSignal("const")
       readonly t RestartUSecMax = ...;
       @org.freedesktop.DBus.Property.EmitsChangedSignal("false")
-      readonly t RestartUSecCurrent = ...;
+      readonly t RestartUSecNext = ...;
```
Comment (bluca) [inline man/org.freedesktop.systemd1.xml:2581]: This is a public interface, so we cannot change it - there needs to be a new one added instead

Comment (bluca) [reply man/org.freedesktop.systemd1.xml:2581]: never mind, it's a new interface

File: src/core/service.c [hunk 2]
```diff
@@ -268,38 +268,36 @@ static void service_start_watchdog(Service *s) {
...
+        /* When the service state is in SERVICE_*_BEFORE_AUTO_RESTART or SERVICE_AUTO_RESTART,
+         * we still need to add 1 to s->n_restarts manually because s->n_restarts is not updated
+         * until a restart job is enqueued. Note that for SERVICE_AUTO_RESTART, that might have been
+         * the case, i.e. s->n_restarts is already increased. But we assume it's not since the time
+         * between job enqueuing and running is usually neglectable compared to the time we'll be sleeping. */
+        n_restarts_next = s->n_restarts + 1;
 
-        /* n_restarts can equal to 0 if no restart has happened nor planned */
-        if (n_restarts <= 1 ||
+        if (n_restarts_next == 1 ||
```
Comment (yuwata) [inline src/core/service.c]: With this change, `n_restart_next` is always positive, so the condition should be OK, but for safety and readability, still I'd like to use `<=` for the comparison.

File: src/core/service.c [hunk 3]
```diff
@@ -268,38 +268,36 @@ static void service_start_watchdog(Service *s) {
...
-        else if (n_restarts > s->restart_steps)
+        else if (n_restarts_next > s->restart_steps)
                 value = s->restart_usec_max;
         else {
                 /* Enforced in service_verify() and above */
                 assert(s->restart_usec_max > s->restart_usec);
 
                 unit = powl(s->restart_usec_max - s->restart_usec, 1.0L / s->restart_steps);
-                value = usec_add(s->restart_usec, (usec_t) powl(unit, n_restarts - 1));
+                value = usec_add(s->restart_usec, (usec_t) powl(unit, n_restarts_next - 1));
```
Comment (yuwata) [inline src/core/service.c]: If you touch the code, then could you merge the two `powl()`?

**Review by yuwata — APPROVED** [review]
LGTM. Minor suggestions.

File: src/core/service.c (hunk 3 above)
Comment (YHNdnzj) [reply src/core/service.c]: Actually I'd like to preserve it for readability 🤔

Comment (yuwata) [reply src/core/service.c]: For readability? I prefer x^yz over (x^y)^z... Or do I misunderstand the implementation?

Comment (YHNdnzj) [reply src/core/service.c]: Oh, you meant to calculate the exponent first. Make sense.

**Review by bluca — APPROVED** [review]

**Pull request merged**